components/network/settings/controller.tsx

```diff
@@ -0,0 +1,243 @@
+      })
+      .catch(handleError)
+      .finally(() => setIsUpdating(false));
+  }
+
+  async function updateNetworkData() {
+    if (isCurrentNetwork) await updateActiveNetwork(true);
+
+    await updateEditingNetwork();
+  }
+
+  useEffect(() => {
+    const logoSize = (details?.fullLogo?.value?.raw?.size || 0) / 1024 / 1024;
+    const iconSize = (details?.iconLogo?.value?.raw?.size || 0) / 1024 / 1024;
+
+    if (logoSize + iconSize >= 1) {
+      setErrorBigImages(true);
+    } else {
+      setErrorBigImages(false);
+    }
+  }, [details?.fullLogo, details?.iconLogo]);
+
+  useEffect(() => {
+    if (!state.Service?.active?.registry?.contractAddress ||
+        !state.currentUser?.walletAddress ||
+        !state.connectedChain?.id) return;
+
+    state.Service?.active
+      .isRegistryGovernor(state.currentUser?.walletAddress)
+      .then(setIsGovernorRegistry);
+  }, [state.currentUser?.walletAddress, state.Service?.active?.registry?.contractAddress, state.connectedChain?.id]);
+
+  useEffect(() => {
+    if(!network) return;
+
+    setTabs([
+      {
+        eventKey: "logo-and-colours",
+        title: t("custom-network:tabs.logo-and-colours"),
+        component: (
+          <NetworkLogoAndColorsSettings
+            network={network}
+            errorBigImages={errorBigImages}
+          />
+        ),
+      },
+      {
+        eventKey: "repositories",
+        title: t("custom-network:tabs.repositories"),
+        component: (
+          <NetworkRepositoriesSettings />
+        ),
+      },
+      {
+        eventKey: "governance",
+        title: t("custom-network:tabs.governance"),
+        component: (
+          <NetworkGovernanceSettings
+            address={network?.networkAddress}
+            tokens={network?.tokens}
+            network={network}
+            updateEditingNetwork={updateEditingNetwork}
+          />
+        ),
+      },
+      {
+        eventKey: "registry",
+        title: t("custom-network:tabs.registry"),
+        component: (
+          <NetworkRegistrySettings isGovernorRegistry={isGovernorRegistry} />
+        ),
+      },
+      {
+        eventKey: "management",
+        title: t("bounty:management.label"),
+        component: (
+          <NetworkManagement bounties={bounties} />
+        )
+      }
+    ])
+  },[
+    network,
+    isGovernorRegistry,
+    networkNeedRegistration,
+    errorBigImages
+  ]);
+
+  return(
+    <MyNetworkSettingsView
+      themePreview={isCurrentNetwork ? colorsToCSS(settings?.theme?.colors) : ""}
+      isNetworkUnregistered={networkNeedRegistration}
+      isWalletConnected={!!state.currentUser?.walletAddress}
+      updateNetworkData={updateNetworkData}
+      networkAddress={network?.networkAddress}
+      tabs={tabs.map(tab => ({
+        label: tab?.title,
+        active: tab?.eventKey === activeTab,
+        onClick: () => setActiveTab(tab?.eventKey)
+      }))}
+      tabsProps={tabs}
+      activeTab={activeTab}
+      isAbleToSave={
```

Comment:
i think create a function for that. help in understanding and future maintenance